### PR TITLE
Fixing Distributions documentation

### LINKED ISSUE #579: Can not sample from Von Mises
```julia
julia> v=VonMises(30,2)
Distributions.VonMises{Float64}(μ=30.0, κ=2.0)

julia> rand(v)
ERROR: MethodError: no method matching quantile(::Distributions.VonMises{Float64}, ::Float64)
Closest candidates are:
  quantile(::Distributions.Distribution{Distributions.Univariate,S<:Distributions.ValueSupport}, ::Real) at /home/elbeltagy/.julia/Distributions/src/univariates.jl:163
  quantile(::Distributions.DiscreteUniform, ::Float64) at /home/elbeltagy/.julia/Distributions/src/univariate/discrete/discreteuniform.jl:114
  quantile(::Distributions.EmpiricalUnivariateDistribution, ::Float64) at /home/elbeltagy/.julia/Distributions/src/empirical.jl:61
  ...
 in rand(::Distributions.VonMises{Float64}) at /home/elbeltagy/.julia/Distributions/src/univariates.jl:64

```

### PR DESCRIPTION
**Pull request by matbesancon**
Badly broken in many places.
Also fixes #579 open for too long (sorry about conflated purposes)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,10 +19,49 @@ A Julia package for probability distributions and associated functions. Particul
...
+not been reported yet on the issues of the repository.
+If not, you can file a new issue, add your version of the package
+which you can get with this command in the Julia REPL:
+```julia
+julia> import Pkg
+julia> Pkg.status("Distributions")
+```
+
+Be exhaustive in your report, give the summary of the bug,
+a Minimum Working Example (MWE), what happens and what you
```
Comment (asinghvi17) [inline README.md]:
```suggestion
a Minimal Working Example (MWE), what happens and what you
```

**Comment by matbesancon** [comment]
Merging in 12 hours if no objection

File: README.md [hunk 2]
```diff
@@ -19,10 +19,49 @@ A Julia package for probability distributions and associated functions. Particul
...
+or [Slack](https://julialang.slack.com), get an invitation
+[here](https://slackinvite.julialang.org/).
+
+2. If you have a bug linked with Distributions, check that it has
+not been reported yet on the issues of the repository.
+If not, you can file a new issue, add your version of the package
+which you can get with this command in the Julia REPL:
+```julia
+julia> import Pkg
+julia> Pkg.status("Distributions")
```
Comment (ararslan) [inline README.md]: I'd just recommend the Pkg REPL mode: `]status Distributions`

File: README.md [hunk 3]
```diff
@@ -19,10 +19,49 @@ A Julia package for probability distributions and associated functions. Particul
...
+```
+
+Be exhaustive in your report, give the summary of the bug,
+a Minimal Working Example (MWE), what happens and what you
+expected to happen.
+
+### Workflow with git and github
```
Comment (ararslan) [inline README.md]:
```suggestion
### Workflow with Git and GitHub
```

File: README.md [hunk 4]
```diff
@@ -19,10 +19,49 @@ A Julia package for probability distributions and associated functions. Particul
...
+
+To contribute to the package, fork the repository on github,
```
Comment (ararslan) [inline README.md]:
```suggestion
To contribute to the package, fork the repository on GitHub,
```

File: README.md (hunk 2 above)
Comment (matbesancon) [reply README.md]: fixed in last commit thanks

**Comment by ararslan** [comment]
Seems fine to me if the docs build and look right.

**Comment by matbesancon** [comment]
yes, in fact they did not build right before this. If you look at the doc site, "stable" is really old, still has some code with "immutable" somewhere, and the tags stopped a long time ago

**Comment by matbesancon** [comment]
Fix #898

File: docs/src/index.md [hunk 5]
```diff
@@ -2,8 +2,9 @@
 
 The *Distributions* package provides a large collection of probabilistic distributions and related functions. Particularly, *Distributions* implements:
 
+* Sampling from distributions
 * Moments (*e.g* mean, variance, skewness, and kurtosis), entropy, and other properties
 * Probability density/mass functions (pdf) and their logarithm (logpdf)
 * Moment generating functions and characteristic functions
```
Comment (Peiffap) [inline docs/src/index.md]: This was already in the docs, but it should be "Moment-generating".

File: src/multivariate/mvnormal.jl [hunk 6]
```diff
@@ -222,7 +229,7 @@ end
 
 ### Show
 
-distrname(d::IsoNormal) = "IsoNormal"    # Note: IsoNormal, etc are just alias names
+distrname(d::IsoNormal)  = "IsoNormal"    # Note: IsoNormal, etc are just alias names
```
Comment (Peiffap) [inline src/multivariate/mvnormal.jl]: Punctuation.

File: src/common.jl [hunk 7]
```diff
@@ -1,10 +1,18 @@
 ## sample space/domain
 
+"""
+`F <: VariateForm` specifies the form of the variate or
+dimension of a sample, univariate (scalar), multivariate (vector), matrix-variate (matrix).
```
Comment (Peiffap) [inline src/common.jl]:
```suggestion
dimension of a sample: univariate (scalar), multivariate (vector) or matrix-variate (matrix).
```

File: docs/src/index.md (hunk 5 above)
Comment (Peiffap) [inline docs/src/index.md]: This was already in the docs, but it should be "Moment-generating".

File: src/multivariate/mvnormal.jl (hunk 6 above)
Comment (matbesancon) [reply src/multivariate/mvnormal.jl]: ?

File: src/common.jl (hunk 7 above)
Comment (matbesancon) [reply src/common.jl]: I didn't want to put it like an exhaustive list here, anyone can subtype `VariateForm` and define their own

File: docs/src/index.md (hunk 5 above)
Comment (matbesancon) [reply docs/src/index.md]: done thanks

File: src/multivariate/mvnormal.jl (hunk 6 above)
Comment (Peiffap) [reply src/multivariate/mvnormal.jl]:
```suggestion
distrname(d::IsoNormal)  = "IsoNormal"    # Note: IsoNormal, etc are just alias names.
```

File: src/common.jl (hunk 7 above)
Comment (Peiffap) [reply src/common.jl]: My bad then :)

**Comment by taqtiqa-mark** [comment]
Hmm, maybe I should make my pull request so we don't duplicate.... or is this PR nearly done or a longer lived effort?

**Comment by matbesancon** [comment]
@taqtiqa-mark this one is merged once CI passes. What's your PR about?

**Comment by taqtiqa-mark** [comment]
Okay, no problem then.  No major overlap.
 - Univariate implementation list #124, #668 
 - Contribution page in docs (how to build docs etc.)
 - travis build stages to build docs per Documenter.jl
Will break them into separate PR to ease rejection ;)

**Comment by simonbyrne** [comment]
Woohoo! this is great

**Pull request merged**